lib/tx_build/set_options.ex

```diff
@@ -0,0 +1,178 @@
+defmodule Stellar.TxBuild.SetOptions do
+  @moduledoc """
+  Sets various configuration options for an account.
+  """
+  alias Stellar.TxBuild.{
+    Flags,
+    OpValidate,
+    OptionalAccount,
+    OptionalAccountID,
+    OptionalFlags,
+    OptionalString32,
+    OptionalSigner,
+    OptionalString32,
+    OptionalWeight,
+    Signer,
+    String32,
+    Weight
+  }
+
+  alias StellarBase.XDR.{OperationBody, OperationType, Operations.SetOptions}
+
+  @behaviour Stellar.TxBuild.XDR
+
+  @type component :: {atom(), any()}
+  @type validation :: {:ok, any()} | {:error, Keyword.t()}
+
+  @type t :: %__MODULE__{
+          inflation_destination: OptionalAccountID.t(),
+          clear_flags: OptionalFlags.t(),
+          set_flags: OptionalFlags.t(),
+          master_weight: OptionalWeight.t(),
+          low_threshold: OptionalWeight.t(),
+          medium_threshold: OptionalWeight.t(),
+          high_threshold: OptionalWeight.t(),
+          home_domain: OptionalString32.t(),
+          signer: OptionalSigner.t(),
+          source_account: OptionalAccount.t()
+        }
+
+  defstruct [
+    :inflation_destination,
+    :clear_flags,
+    :set_flags,
+    :master_weight,
+    :low_threshold,
+    :medium_threshold,
+    :high_threshold,
+    :home_domain,
+    :signer,
+    :source_account
+  ]
+
+  @impl true
+  def new(args, opts \\ [])
+
+  def new(args, _opts) when is_list(args) do
+    inf_dest = Keyword.get(args, :inflation_destination)
+    clear_flags = Keyword.get(args, :clear_flags)
+    set_flags = Keyword.get(args, :set_flags)
+    master_weight = Keyword.get(args, :master_weight)
+    low_threshold = Keyword.get(args, :low_threshold)
+    med_threshold = Keyword.get(args, :medium_threshold)
+    high_threshold = Keyword.get(args, :high_threshold)
+    home_domain = Keyword.get(args, :home_domain)
+    signer = Keyword.get(args, :signer)
+    source_account = Keyword.get(args, :source_account)
+
+    with {:ok, inf_dest} <- validate(:account_id, {:inflation_destination, inf_dest}),
+         {:ok, clear_flags} <- validate(:flags, {:clear_flags, clear_flags}),
+         {:ok, set_flags} <- validate(:flags, {:set_flags, set_flags}),
+         {:ok, master_weight} <- validate(:weight, {:master_weight, master_weight}),
+         {:ok, low_threshold} <- validate(:weight, {:low_threshold, low_threshold}),
+         {:ok, med_threshold} <- validate(:weight, {:medium_threshold, med_threshold}),
+         {:ok, high_threshold} <- validate(:weight, {:high_threshold, high_threshold}),
+         {:ok, home_domain} <- validate(:string32, {:home_domain, home_domain}),
+         {:ok, signer} <- validate(:signer, {:signer, signer}),
+         {:ok, source_account} <- validate(:account, {:source_account, source_account}) do
```

Comment:
@luishurtado following your suggestion, validations were implemented as private functions. What do you think? looks it better?

Note: Common validations need to be refactored.

Best!